editor.planx.uk/src/@planx/components/shared/Schema/model.ts

```diff
@@ -36,70 +36,104 @@ interface ChecklistInput {
 /**
  * As above, we need a simplified validation schema for QuestionsInputs
  */
-const questionInputValidationSchema = (data: QuestionInput) =>
+
+export const questionInputValidationSchema = ({
+  data,
+  required,
+}: FieldValidationSchema<QuestionInput>) =>
   string()
-    .oneOf(data.options.map((option) => option.data.val || option.data.text))
-    .required("Select your answer before continuing");
+    .when([], {
+      is: () => required,
+      then: string().required("Select your answer before continuing"),
+      otherwise: string().notRequired(),
+    })
+    .test("isValidOption", "Invalid selection", (value) => {
+      if (!value) return true;
+
+      // Check validity regardless of "required" status
+      return data.options.some(
+        (option) => value === (option.data.val || option.data.text),
+      );
+    });
+
+/**
+ * Describes the input of function which returns a Yup validation schema for a field
+ */
+export interface FieldValidationSchema<T extends Omit<Field["data"], "fn">> {
+  data: T;
+  required: boolean;
+}
 
-const mapValidationSchema = ({ mapOptions }: MapField["data"]) =>
+export const mapInputValidationSchema = ({
+  data: { mapOptions },
+  required,
+}: FieldValidationSchema<MapInput>) =>
   array()
-    .required()
+    .when([], {
+      is: () => required,
+      then: array().required(),
+      otherwise: array().notRequired(),
+    })
     .test({
       name: "atLeastOneFeature",
       message: `Draw at least one ${
         mapOptions?.drawType?.toLocaleLowerCase() || "feature"
       } on the map`,
       test: (features?: Array<Feature>) => {
-        return Boolean(features && features?.length > 0);
+        if (!required && !features?.length) return true;
+        if (!features || !features.length) return false;
+
+        const isGeoJSON = (input: Feature) => input?.type === "Feature";
```

Comment:
Pretty crude, there's probably some good libraries for GeoJson validation (turf probably has something) but I think this is fine for now. 

I was hitting an issue where any array (e.g an array of strings) that I passed in met the validation requirements so I couldn't test this.